Documentation/design-docs/COM-activation.md

```diff
@@ -0,0 +1,132 @@
+# COM Activation for .NET Core on Windows
+
+## Purpose
+
+In order to more fully support the vast number of existing .NET Framework users in their transition to .NET Core, support of the COM activation scenario in .NET Core is required. Without this support it is not possible for many .NET Framework consumers to even consider transitioning to .NET Core. The intent of this document is to describe aspects of COM activation for a .NET class written for .NET Core. This support includes but is not limited to activation scenarios such as the [`CoCreateInstance()`](https://docs.microsoft.com/en-us/windows/desktop/api/combaseapi/nf-combaseapi-cocreateinstance)API in C/C++ or from within a [Windows Script Host](https://docs.microsoft.com/en-us/windows/desktop/com/using-com-objects-in-windows-script-host) instance.
+
+COM activation in this document is currently limited to in-proc scenarios. Scenarios involving out-of-proc COM activation are deferred.
+
+### Requirements
+
+* Discover all installed versions of .NET Core.
+* Load the appropriate version of .NET Core for the class if a .NET Core instance is not running, or validate the currently existing .NET Core instance can satisfy the class requirement.
+* Instantiate an instance of the class and return a pointer to an [`IUnknown`](https://docs.microsoft.com/en-us/windows/desktop/api/unknwn/nn-unknwn-iunknown) that represents the newly constructed class.
+* Support the discrimination of concurrently loaded CLR versions.
+
+### Environment Matrix
+
+The following list represents an exhaustive activation matrix.
+
+| Server | Client | Current Support |
+| --- | --- | :---: |
+| COM* | .NET Core | Yes |
+| .NET Core | COM* | No |
+| .NET Core | .NET Core | No |
+| .NET Framework | .NET Core | No |
+| .NET Core | .NET Framework | No |
+
+\* 'COM' is used to indicate any COM environment (e.g. C/C++) other than .NET.
+
+## Design
+
+One of the basic issues with the activation of a .NET class within a COM environment is the loading or discovery of an appropriate CLR instance. The .NET Framework addressed this issue through a shim library (described below). The .NET Core scenario has different requirements and limitations on system impact and as such an identical solution may not be optimal or tenable.
```

Comment:
> The .NET Core scenario has different requirements and limitations on system impact

Could you list some of these difference in this design doc?  so customer can understand better why we design this way.